addons/sale/models/sale_order.py

```diff
@@ -520,6 +520,12 @@ def create(self, vals):
         result = super(SaleOrder, self).create(vals)
         return result
 
+    def write(self, vals):
+        if 'commitment_date' in vals:
+            for order in self.sudo():
+                order.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel')).write({'date_deadline': vals['commitment_date'] or order.expected_date})
```

Comment:
```suggestion
        if 'commitment_date' in vals and vals['commitment_date']:
            for order in self:
                order.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel')).sudo().write({'date_deadline': vals['commitment_date']})
```

Also, check commitment_date have value.
No need of or in write because we already check in first step